warehouse/query-core/src/main/java/datawave/query/planner/scanhints/IvaratorScanHint.java

```diff
@@ -0,0 +1,38 @@
+package datawave.query.planner.scanhints;
+
+import org.apache.commons.jexl3.parser.JexlNode;
+
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.jexl.NodeTypeCount;
+
+/**
+ * Detect ivarators and set the executor scan hint to ivarator
+ */
+public class IvaratorScanHint extends ExecutorScanHintRule {
```

Comment:
Could you add a test that runs through several variations of queries to prove some positive and negative cases? It doesn't need to be as expansive as the RewriteNegationsVisitorTest, but that gives you an idea of what I'm looking for.